backend/fms_core/services/experiment_run_info.py

```diff
@@ -0,0 +1,353 @@
+                row.chip_seq_mark = library.library_selection.target
+
+    return row
+
+def _convert_flowcell_coordinate_to_lane_number(coord: str, container: Container) -> int:
+    '''Converts a freezeman coordinate (eg. A01) to an integer flowcell lane number.'''
+
+    container_spec = CONTAINER_KIND_SPECS[container.kind]
+    if container_spec is None:
+        raise Exception(f'Cannot convert coord {coord} to lane number. No ContainerSpec found for container kind "{container.kind}".')
+
+    lane = convert_alpha_digit_coord_to_ordinal(coord, container_spec.coordinate_spec)
+    return lane
+
+def _find_library_prep(library: Library) -> Union[ProcessMeasurement, None]:
+    ''' 
+    Given a library, find the Library Preparation process that was used to create the
+    library, if any, to extract the library preparation properties (eg. Library Kit).
+
+    Note that not all libraries will have a Library Preparation measurement,
+    since they can be imported directly into freezeman without passing the
+    Library Preparation step.
+    '''
+    
+    library_prep: Union[ProcessMeasurement, None] = None
+    try:
+        library_to_find : Library = library
+
+        if library.library_selection is not None:
+            # If a library was captured then that generated a new library instance.
+            # We have to find the previous instance, on which the Library Prep was run.
+            try:
+                capture_protocol = Protocol.objects.get(name="Library Capture")
+                capture_lineage: SampleLineage = SampleLineage.objects.get(
+                    process_measurement__process__protocol_id=capture_protocol.pk,
+                    child__derived_samples__library__id=library.pk
+                )
+                #library_to_find = Library.objects.get(derived_sample__biosample_id=derived_sample.biosample.id)
+                # Now find the derived sample in the parent that contains the same biosample id as our library.
+                library_to_find = Library.objects.get(derived_sample__samples=capture_lineage.parent, derived_sample__biosample=library.derived_sample.biosample)
+                
+            except:
+                pass
+
+        # Find the library preparation process measurement
+        protocol: Protocol = Protocol.objects.get(name="Library Preparation")
+        lineage: SampleLineage = SampleLineage.objects.get(
+            process_measurement__process__protocol_id=protocol.pk,
+            child__derived_samples__library__id=library_to_find.pk)
+        library_prep = lineage.process_measurement
+    except SampleLineage.DoesNotExist:
+        pass
+    except SampleLineage.MultipleObjectsReturned:
+        pass
+
+    return library_prep
+
+
+def _find_library_capture_process(derived_sample: DerivedSample) -> Union[Process, None]:
+    '''If this is a captured library, find the process measurement from the capture step.'''
+    if derived_sample.library is None:
+        return None
```

Comment:
I think we need to give a thought about the error handling structure here and other sub-functions. For exemple at the moment it is not an option for us to deliver a run info file for a sample without a library, but it is possible for a library not to have a capture (library_selection). Both will see a return value of None. At this level it is not possible to make a decision about what to do about these value, so i would recommend this function just make sure there is or not a capture and pass to the parent the task of making sure if this make sense that there is a library. Some future run (Axiom or other type of processing) are likely to involve samples without library we may want to reuse these functions between different platforms. The capture is not quite reusable without library but the idea is to have a error return value that can be interpreted as either an unexpected error OR an empty value not both.